File: app/src/main/java/br/com/jaysonsabino/desafioandroidpopcode/services/swapi/PeopleService.java
package br.com.jaysonsabino.desafioandroidpopcode.services.swapi;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface PeopleService {

    @GET("people/")
    Call<PeopleListResponseDTO> getList(@Query("page") int page);
}

File: app/src/main/java/com/arthurgonzaga/wikistarwars/api/services/PeopleService.kt
package com.arthurgonzaga.wikistarwars.api.services

import com.arthurgonzaga.wikistarwars.api.responses.PageResponse
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface PeopleService {

    @GET("people")
    fun getPeoplePage(
        @Query("page") page: Int,
        @Query("search") query: String = ""
    ): Single<PageResponse>

}